### PR TITLE
remove or merge redundant or non needed tests

### PR DESCRIPTION
**Pull request by rplevka**
first step in the refactoring of CLI discovered tests.

### CODE REVIEW

File: tests/foreman/cli/test_discoveredhost.py [hunk 1]
```diff
@@ -130,72 +130,6 @@ def test_positive_pxe_based_discovery(self):
             host = self._assertdiscoveredhost(hostname)
             self.assertIsNotNone(host)
 
-    @pytest.mark.tier3
-    def test_positive_pxe_less_with_dhcp_unattended(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:134]: this is already contained as a step in
`test_positive_provision_pxeless_bios_syslinux`

File: tests/foreman/cli/test_discoveredhost.py [hunk 2]
```diff
@@ -271,120 +205,6 @@ def test_positive_provision_pxeless_bios_syslinux(self):
             with self.assertRaises(CLIReturnCodeError):
                 DiscoveredHost.info({'id': discovered_host['id']})
 
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_provision_pxeless_uefi_grub(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:276]: UEFI-GRUB is an unsupported combo

File: tests/foreman/cli/test_discoveredhost.py [hunk 3]
```diff
@@ -271,120 +205,6 @@ def test_positive_provision_pxeless_bios_syslinux(self):
...
-            7. Ensure the entry from discovered host list disappeared
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: High
-        """
-
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_provision_pxeless_uefi_grub2(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:314]: Bootloader doesn't matter in the pxeless scenario, as a `kexec` is used

File: tests/foreman/cli/test_discoveredhost.py [hunk 4]
```diff
@@ -271,120 +205,6 @@ def test_positive_provision_pxeless_bios_syslinux(self):
...
-            7. Ensure the entry from discovered host list disappeared
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: High
-        """
-
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_provision_pxeless_uefi_grub2_secureboot(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:352]: Secureboot is not supported, also the firwmare doesn't matter in pxeless scenario

File: tests/foreman/cli/test_discoveredhost.py [hunk 5]
```diff
@@ -475,55 +295,6 @@ def test_positive_provision_pxe_host_with_bios_syslinux(self):
             with self.assertRaises(CLIReturnCodeError):
                 DiscoveredHost.info({'id': discovered_host['id']})
 
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_provision_pxe_host_with_uefi_grub(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:480]: not supported scenario

File: tests/foreman/cli/test_discoveredhost.py [hunk 6]
```diff
@@ -574,79 +345,9 @@ def test_positive_provision_pxe_host_with_uefi_grub2(self):
         :CaseImportance: High
         """
 
-    @pytest.mark.stubbed
     @pytest.mark.tier3
-    def test_positive_provision_pxe_host_with_uefi_grub2_sb(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:579]: secure boot not supported

File: tests/foreman/cli/test_discoveredhost.py [hunk 7]
```diff
@@ -574,79 +345,9 @@ def test_positive_provision_pxe_host_with_uefi_grub2(self):
...
-            8. Ensure the host is provisioned with correct attributes
-            9. Ensure the entry from discovered host list disappeared
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: High
-        """
-
-    @pytest.mark.tier3
-    def test_positive_delete_pxeless_host(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:628]: There's no difference between removing a pxeless host or a pxe-based host - it's a discovered host entity in foreman.

File: tests/foreman/cli/test_discoveredhost.py [hunk 8]
```diff
@@ -758,20 +459,6 @@ def test_positive_auto_provision_pxeless_host(self):
         :CaseImportance: Critical
         """
 
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_list_discovered_host(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:763]: we already do this as a part of
`test_positive_provision_pxeless_bios_syslinux`

File: tests/foreman/cli/test_discoveredhost.py [hunk 9]
```diff
@@ -758,20 +459,6 @@ def test_positive_auto_provision_pxeless_host(self):
...
-        :expectedresults: Host should be discovered and listed with names.
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: High
-        """
-
     @pytest.mark.stubbed
     @pytest.mark.tier3
     def test_positive_assign_discovery_manager_role(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:464]: i'll most probably use non-admin user with discovery roles on all tests after refactoring. I might parameterize the tests to use each role.

File: tests/foreman/cli/test_discoveredhost.py [hunk 10]
```diff
@@ -130,72 +130,6 @@ def test_positive_pxe_based_discovery(self):
...
-        :CaseImportance: Critical
-        """
-        with LibvirtGuest(boot_iso=True) as pxe_less_host:
-            hostname = pxe_less_host.guest_name
-            host = self._assertdiscoveredhost(hostname)
-            self.assertIsNotNone(host)
-
-    @pytest.mark.stubbed
-    @pytest.mark.tier3
-    def test_positive_custom_facts_pxeless_discovery(self):
```
Comment (rplevka) [inline tests/foreman/cli/test_discoveredhost.py:155]: there's no difference in listing facts of pxe-based or pxeless discovered host, merging both to a single test:
`test_positive_list_facts`
where both builtin and custom facts are to be checked.

**Review by lvrtelov — APPROVED** [review]
LGTM. Thanks for adding those comments, Roman!

**Review by tstrych — APPROVED** [review]
ACK, please look at code review failiure

```
tests/foreman/cli/test_discoveredhost.py
========================================

DiscoveredTestCase::test_positive_list_facts:511
------------------------------------------------

* Docstring should have at least caseautomation, casecomponent, caseimportance, caselevel, id, requirement, testtype, upstream token(s)
* Docstring has RST parsing issues. RST parser messages:

  * Unexpected indentation.

      :id: 2c65925c-05d9-4f6d-b1b7-1fa1492c95a8
      :Setup:
    >     1. Provisioning should be configured
          2. Host is already discovered
    ```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Awesome! Just need to resolve the failures @tstrych noted above.

File: tests/foreman/cli/test_discoveredhost.py [hunk 11]
```diff
@@ -780,12 +467,12 @@ def test_positive_assign_discovery_manager_role(self):
         :id: f694c361-5fbb-4c3a-b2ff-6dfe6ea14820
 
         :expectedresults: User should be able to view, provision, edit and
-            destroy one or more discovered host as well view, create_new, edit,
-            execute and delete discovery rules.
+        destroy one or more discovered host as well view, create_new, edit,
```
Comment (swadeley) [inline tests/foreman/cli/test_discoveredhost.py:470]:
```suggestion
        destroy one or more discovered hosts as well as view, create_new, edit,
```

File: tests/foreman/cli/test_discoveredhost.py [hunk 12]
```diff
@@ -796,11 +483,11 @@ def test_positive_assign_discovery_role(self):
         :id: 873e8411-563d-4bf9-84ce-62e522410cfe
 
         :expectedresults: User should be able to list, provision, and destroy
-            one or more discovered host
+        one or more discovered host
```
Comment (swadeley) [inline tests/foreman/cli/test_discoveredhost.py:486]:
```suggestion
        one or more discovered hosts
```

**Review by swadeley — APPROVED** [review]
ACK

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**